pkg/apis/sessions/session_state.go

```diff
@@ -233,7 +238,7 @@ func (s *SessionState) validate() error {
 	}
 
 	empty := new(SessionState)
-	if *s == *empty {
+	if reflect.DeepEqual(*s, *empty) {
```

Comment:
The addition of the slice broke the `==` I went the quick route, but can write an Equals function here if desired.

Reply:
Let me dig into what messagePack is doing when it gets a valid header but garbage data (hence unmarshaling an empty SessionState) Since this logic happens every request, I'd aim for pure speed to not be a bottleneck.  Not positive if reflect will slow us down too much?

`nil` for slice on both sides of the compare doesn't work with `==`?

Reply:
I just get a compile erorr saying `==` is not implemented once the slice is added to the struct.